webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ExpectedReportsRepository.scala

```diff
@@ -109,6 +110,13 @@ trait FindExpectedReportRepository {
    */
   def findCurrentNodeIdsForRule(ruleId: RuleId, nodeIds: Set[NodeId]): IOResult[Set[NodeId]]
 
+  /**
+   * Return node ids associated to the rule (based on expectedreports (the one still pending)) for this Rule,
+   * only limited on the nodeIds in parameter (used when cache is incomplete)
+   */
+  def findCurrentNodeIdsForDirective(ruleId: DirectiveId, nodeIds: Set[NodeId]): IOResult[Set[NodeId]]
+  def findCurrentNodeIdsForDirective(ruleId: DirectiveId): IOResult[Set[NodeId]]
```

Comment:
you should not use the same function name for different signatures, it creates problems latter on (inference, refactoring, search not working as expected, etc). 
At least, each and every time I thought it was a good idea, I got bitten down the path latter on

Reply:
Actually, given the code it could be that `nodeIds` is an option (or have the semantic "if set is empty, ignore id") and just add the `and` fragment when it's not empty

Reply:
What is complex is that to keep the current behaviour:
If some nodes ids => Make the request with a where condition on nodeIds
If none => Make the request without where on node ids
If Some empty set => Don't make request 

I'm ok that None and some(Empty) makes sense to not have the and  So i need to look the semantic of the empty Set here

And yes i would totally love rewriting this query using more doobie fragments and tools 

So i may do another PR to improve this